cdc/sink/mq/dispatcher/event_router.go

```diff
@@ -249,17 +250,32 @@ func getPartitionDispatcher(
 
 // getTopicDispatcher returns the topic dispatcher for a specific topic rule (aka topic expression).
 func getTopicDispatcher(
-	ruleConfig *config.DispatchRule, defaultTopic string,
+	ruleConfig *config.DispatchRule, defaultTopic string, protocol string,
 ) (topic.Dispatcher, error) {
 	if ruleConfig.TopicRule == "" {
 		return topic.NewStaticTopicDispatcher(defaultTopic), nil
 	}
 
 	// check if this rule is a valid topic expression
 	topicExpr := topic.Expression(ruleConfig.TopicRule)
-	err := topicExpr.Validate()
-	if err != nil {
-		return nil, err
+
+	if protocol != "" {
+		var p config.Protocol
+		if err := p.FromString(protocol); err != nil {
```

Comment:
It should have been done in the configuration check?

Reply:
That's the sad part. The protocol is parsed and simply used to new encoder config, not stored inside replica config. So at the time of newMqSink, the parsed protocol is not available. I don't understand why actually.

Here, I re-parse the protocol to ultilize the parsing logic in case anything might change.